src/main/kotlin/com/linchpino/ai/service/impl/ResumeService.kt

```diff
@@ -0,0 +1,72 @@
+package com.linchpino.ai.service.impl
+
+import com.linchpino.ai.model.Resume
+import com.linchpino.ai.repository.ResumeRepository
+import com.linchpino.core.exception.ErrorCode
+import com.linchpino.core.exception.LinchpinException
+import org.apache.pdfbox.Loader
+import org.apache.pdfbox.text.PDFTextStripper
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Service
+import java.io.File
+import java.io.IOException
+import java.util.*
+
+@Service
+class ResumeService(private val resumeRepository: ResumeRepository) {
+    private val logger: Logger = LoggerFactory.getLogger(ResumeService::class.java)
+
+    @Value("\${spring.ai.roadmap-attempt-limit}")
+    private val roadmapAttemptLimit = 0
+
+    fun create(file: File?): Resume {
+        requireNotNull(file) { "File cannot be null" }
+        val lines = readLines(file)
+        val email = Resume.findEmail(lines)
+        if (isRoadmapAttemptLimitReached(email)) {
+            throw LinchpinException(ErrorCode.TOO_MANY_ATTEMPT, "Roadmap attempt limit reached for email: {}", email)
+        }
+        return Resume(email, lines)
+    }
+
+    fun save(resume: Resume): Resume {
+        Objects.requireNonNull(resume, "Resume cannot be null")
+        Objects.requireNonNull(resume.email, "Email cannot be null")
+        return resumeRepository.save(resume)
+    }
+
+    fun findById(id: Long): Resume? {
+        return resumeRepository.findById(id).orElse(null)
+    }
+
+    fun findAll(): MutableList<Resume?> {
+        return resumeRepository.findAll()
+    }
+
+    private fun isRoadmapAttemptLimitReached(email: String): Boolean {
+        return resumeRepository.countAllByEmail(email) >= roadmapAttemptLimit
+    }
+
+    /**
+     * Read the lines from the PDF file
+     *
+     * @param file File
+     * @return List of lines
+     */
+    private fun readLines(file: File): List<String> {
+        try {
+            Loader.loadPDF(file).use { document ->
+                logger.info("Reading the file: {}", file.name)
+                val textStripper = PDFTextStripper()
+                return ArrayList(
+                    Arrays.asList(
+                        *textStripper.getText(document).split("\n".toRegex()).dropLastWhile { it.isEmpty() }
```

Comment:
split returns list. there is no need to bring Java's verbosity to Kotlin. No spreading and converting to byte array and again converting bytearray to list is needed.
you can just: 
return PDFTextStripper().getText(document).split("\n".toRegex()).dropLastWhile { it.isEmpty() }